lib/hydra/pcdm/models/concerns/object_behavior.rb

```diff
@@ -79,6 +75,30 @@ def contains= files
       super(files)
     end
 
+    # Returns directly contained files that have the requested RDF Type
+    # @param [RDF::URI] uri for the desired Type
+    # @example
+    #   filter_files_by_type(::RDF::URI("http://pcdm.org/ExtractedText"))
+    def filter_files_by_type uri
+      self.files.reject do |file|
```

Comment:
@jcoyne regarding projecthydra-labs/hydra-works#124 how would I rewrite this (or line 93) to avoid iterating?  Is there a way to query directly for files with a specific RDF type?